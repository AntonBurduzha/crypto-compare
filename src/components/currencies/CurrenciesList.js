// @flow
import * as React from 'react';
import { Row, Col, Card } from 'antd';
import { CRYPRO_COMPARE_OLD_API } from '../../constants';
import type { Currency } from '../../types/entities';
import type { Element } from 'react';

type Props = {
  list: Array<Currency>
};

const CurrenciesList = (props: Props): Element<any> =>
  <Row type="flex" justify="start" gutter={16} style={{margin: '0', padding: '0 20px' }}>
    { props.list.map((item: Currency): Element<any> => {
        return (
          <Col span={3} key={item.Id} style={{ margin: '10px 0', padding: '0' }}>
            <Card
              hoverable
              style={{ width: 140, textAlign: 'center' }}
              cover={
                <a href={`${CRYPRO_COMPARE_OLD_API}${item.Url}`} target="_blank">
                  <img alt={item.FullName} src={`${CRYPRO_COMPARE_OLD_API}${item.ImageUrl}`}/>
                </a>
              }
            >
              <Card.Meta title={item.Symbol} description={item.CoinName}/>
            </Card>
          </Col>
        );
      })
    }
  </Row>

export default CurrenciesList;
