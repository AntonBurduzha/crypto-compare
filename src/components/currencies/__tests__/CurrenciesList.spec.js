import React from 'react';
import { Card } from 'antd';
import { shallow } from 'enzyme';
import CurrenciesList from '../CurrenciesList';

describe('CurrenciesList component', () => {
  const props = {
    list: [
      { Url: '/0/', Id: '0', FullName: 'Anton', ImageUrl: '/qwe', Symbol: 'A', CoinName: 'ACoin' },
      { Url: '/1/', Id: '1', FullName: 'Anto', ImageUrl: '/ewq', Symbol: 'B', CoinName: 'BCoin' },
      { Url: '/2/', Id: '2', FullName: 'Ant', ImageUrl: '/rty', Symbol: 'C', CoinName: 'CCoin' },
      { Url: '/3/', Id: '3', FullName: 'An', ImageUrl: '/yrt', Symbol: 'D', CoinName: 'DCoin' },
      { Url: '/4/', Id: '4', FullName: 'A', ImageUrl: '/zxc', Symbol: 'E', CoinName: 'ECoin' },
    ]
  }
  const wrap = shallow(<CurrenciesList {...props}/>);

  it('renders without crashing', () => {
    expect(wrap).toBeDefined();
  });

  it('contains 5 Cards', () => {
    expect(wrap.find(Card)).toHaveLength(5);
  });

  it('contains 0 Cards', () => {
    wrap.setProps({ list: [] });
    expect(wrap.find(Card)).toHaveLength(0);
  });

  it('contains 1 Card with Meta', () => {
    wrap.setProps({
      list: [{ Url: '/0/', Id: '0', FullName: 'Anton', ImageUrl: '/qwe', Symbol: 'A', CoinName: 'ACoin' }]
    });
    expect(wrap.find(Card.Meta)).toHaveLength(1);
    expect(wrap.find(Card.Meta).props().title).toEqual('A');
    expect(wrap.find(Card.Meta).props().description).toEqual('ACoin');
  });

  it('matches snapshot', () => {
    expect(wrap).toMatchSnapshot();
  });
});
