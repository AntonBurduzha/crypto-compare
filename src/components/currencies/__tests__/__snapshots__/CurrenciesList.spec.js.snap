// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrenciesList component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrenciesList
    list={
        Array [
            Object {
              "CoinName": "ACoin",
              "FullName": "Anton",
              "Id": "0",
              "ImageUrl": "/qwe",
              "Symbol": "A",
              "Url": "/0/",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Col
          span={3}
          style={
                    Object {
                              "margin": "10px 0",
                              "padding": "0",
                            }
          }
>
          <Card
                    cover={
                              <a
                                        href="https://www.cryptocompare.com/0/"
                                        rel="noopener"
                                        target="_blank"
                              >
                                        <img
                                                  alt="Anton"
                                                  src="https://www.cryptocompare.com/qwe"
                                        />
                              </a>
                    }
                    hoverable={true}
                    style={
                              Object {
                                        "textAlign": "center",
                                        "width": 140,
                                      }
                    }
          >
                    <Unknown
                              description="ACoin"
                              title="A"
                    />
          </Card>
</Col>,
      ],
      "gutter": 16,
      "justify": "start",
      "style": Object {
        "margin": "0",
        "padding": "0 20px",
      },
      "type": "flex",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <Card
            cover={
                        <a
                                    href="https://www.cryptocompare.com/0/"
                                    rel="noopener"
                                    target="_blank"
                        >
                                    <img
                                                alt="Anton"
                                                src="https://www.cryptocompare.com/qwe"
                                    />
                        </a>
            }
            hoverable={true}
            style={
                        Object {
                                    "textAlign": "center",
                                    "width": 140,
                                  }
            }
>
            <Unknown
                        description="ACoin"
                        title="A"
            />
</Card>,
          "span": 3,
          "style": Object {
            "margin": "10px 0",
            "padding": "0",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              description="ACoin"
              title="A"
/>,
            "cover": <a
              href="https://www.cryptocompare.com/0/"
              rel="noopener"
              target="_blank"
>
              <img
                            alt="Anton"
                            src="https://www.cryptocompare.com/qwe"
              />
</a>,
            "hoverable": true,
            "style": Object {
              "textAlign": "center",
              "width": 140,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "ACoin",
              "title": "A",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Col
            span={3}
            style={
                        Object {
                                    "margin": "10px 0",
                                    "padding": "0",
                                  }
            }
>
            <Card
                        cover={
                                    <a
                                                href="https://www.cryptocompare.com/0/"
                                                rel="noopener"
                                                target="_blank"
                                    >
                                                <img
                                                            alt="Anton"
                                                            src="https://www.cryptocompare.com/qwe"
                                                />
                                    </a>
                        }
                        hoverable={true}
                        style={
                                    Object {
                                                "textAlign": "center",
                                                "width": 140,
                                              }
                        }
            >
                        <Unknown
                                    description="ACoin"
                                    title="A"
                        />
            </Card>
</Col>,
        ],
        "gutter": 16,
        "justify": "start",
        "style": Object {
          "margin": "0",
          "padding": "0 20px",
        },
        "type": "flex",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <Card
              cover={
                            <a
                                          href="https://www.cryptocompare.com/0/"
                                          rel="noopener"
                                          target="_blank"
                            >
                                          <img
                                                        alt="Anton"
                                                        src="https://www.cryptocompare.com/qwe"
                                          />
                            </a>
              }
              hoverable={true}
              style={
                            Object {
                                          "textAlign": "center",
                                          "width": 140,
                                        }
              }
>
              <Unknown
                            description="ACoin"
                            title="A"
              />
</Card>,
            "span": 3,
            "style": Object {
              "margin": "10px 0",
              "padding": "0",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown
                description="ACoin"
                title="A"
/>,
              "cover": <a
                href="https://www.cryptocompare.com/0/"
                rel="noopener"
                target="_blank"
>
                <img
                                alt="Anton"
                                src="https://www.cryptocompare.com/qwe"
                />
</a>,
              "hoverable": true,
              "style": Object {
                "textAlign": "center",
                "width": 140,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "ACoin",
                "title": "A",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
